homeassistant/components/systemmonitor/sensor.py

```diff
@@ -158,183 +181,289 @@ def check_required_arg(value):
+        if data.argument in counters:
+            counter = counters[data.argument][IO_COUNTER[type_]]
+            state = round(counter / 1024 ** 2, 1)
+        else:
+            state = None
+    elif type_ in ["packets_out", "packets_in"]:
+        counters = psutil.net_io_counters(pernic=True)
+        if data.argument in counters:
+            state = counters[data.argument][IO_COUNTER[type_]]
+        else:
+            state = None
+    elif type_ in ["throughput_network_out", "throughput_network_in"]:
+        counters = psutil.net_io_counters(pernic=True)
+        if data.argument in counters:
+            counter = counters[data.argument][IO_COUNTER[type_]]
+            now = dt_util.utcnow()
+            if data.value and data.value < counter:
+                state = round(
+                    (counter - data.value)
+                    / 1000 ** 2
+                    / (now - (data.update_time or now)).seconds,
+                    3,
+                )
             else:
-                self._state = None
-        elif self.type == "last_boot":
-            # Only update on initial setup
-            if self._state is None:
-                self._state = dt_util.as_local(
-                    dt_util.utc_from_timestamp(psutil.boot_time())
-                ).isoformat()
-        elif self.type == "load_1m":
-            self._state = round(os.getloadavg()[0], 2)
-        elif self.type == "load_5m":
-            self._state = round(os.getloadavg()[1], 2)
-        elif self.type == "load_15m":
-            self._state = round(os.getloadavg()[2], 2)
-
-    @staticmethod
-    def read_cpu_temperature():
-        """Attempt to read CPU / processor temperature."""
-        temps = psutil.sensors_temperatures()
-
-        for name, entries in temps.items():
-            for i, entry in enumerate(entries, start=1):
-                # In case the label is empty (e.g. on Raspberry PI 4),
-                # construct it ourself here based on the sensor key name.
-                _label = f"{name} {i}" if not entry.label else entry.label
-                if _label in CPU_SENSOR_PREFIXES:
-                    return round(entry.current, 1)
+                state = None
+            update_time = now
+            value = counter
+        else:
+            state = None
+    elif type_ in ["ipv4_address", "ipv6_address"]:
+        addresses = psutil.net_if_addrs()
+        if data.argument in addresses:
+            for addr in addresses[data.argument]:
+                if addr.family == IF_ADDRS_FAMILY[type_]:
+                    state = addr.address
+        else:
+            state = None
+    elif type_ == "last_boot":
+        # Only update on initial setup
+        if data.state is None:
+            state = dt_util.as_local(
+                dt_util.utc_from_timestamp(psutil.boot_time())
+            ).isoformat()
+        else:
+            state = data.state
+    elif type_ == "load_1m":
```

Comment:
Should could probably be refactored to only call `os.getloadavg()` once instead of 3x if they have all 3, but I didn't want to expand the scope